bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/db/JdbcBaseDAO.java

```diff
@@ -258,18 +244,23 @@ public Properties getConnectionProperties() {
      * ITEMS DAOs *
      **************/
     public Integer doPingDB() {
-        return Yank.queryScalar(sqlPingDB, Integer.class, null);
+        @Nullable
+        Integer result = Yank.queryScalar(sqlPingDB, Integer.class, null);
+        return result;
```

Comment:
Hmm, I was trying to fix this:
`JdbcBaseDAO.java:[247,26] Unsafe interpretation of method return type as '@NonNull' based on substitution 'T=java.lang.@NonNull Integer'. Declaring type 'org.knowm.yank.Yank' doesn't seem to be designed with null type annotations in mind`

If I add a null check before returning, I get a warning that result cannot be null. Now I'm in doubt if this is actually correct. Current solution is probably the minimum required for preserving existing runtime behavior. Another similar approach might be:

```java
        Integer result = (Integer) Yank.queryScalar(sqlPingDB, Integer.class, null);
```

This could also potentially hide problems in the same way if actually the method can return null.

Do you have any proposals which could be done without significantly increasing scope of this PR?

Reply:
It seems that Yank will internally handles exceptions and silently logs them without rethrowing - so definitely null can be returned, thus my changes are wrong: https://github.com/knowm/Yank/blob/develop/README.md#handle-exceptions

So I think nullness has to be enforced - perhaps something like this:

```java
    public Integer doPingDB() {
        final @Nullable Integer result = Yank.queryScalar(sqlPingDB, (Class<@Nullable Integer>) Integer.class, null);
        return result != null ? result : 0;
    }
```

WDYT? Of course some of the other methods will need to return `@Nullable`.

Reply:
Fixed, please check latest commit, 93d79dc84cb6aaae94b15fde1fce05dd20ee11b6.